src/integration-test/java/com/commercetools/sync/integration/ctpprojectsource/producttypes/ProductTypeWithNestedAttributeSyncIT.java

```diff
@@ -116,6 +123,75 @@ void sync_WithEmptyTargetProject_ShouldReturnProperStatistics() {
                 + " of NestedType attribute definition(s) referencing a missing product type).");
   }
 
+  @Test
+  void sync_WithProductTypeReferencingItselfAsAttribute_ShouldCreateProductType() {
+    // preparation
+    final AttributeDefinitionDraft nestedTypeAttr =
+        AttributeDefinitionDraftBuilder.of()
+            .name("selfReferenceAttr")
+            .label(LocalizedString.ofEnglish("selfReferenceAttr"))
+            .type(
+                AttributeSetTypeBuilder.of()
+                    .elementType(
+                        AttributeSetTypeBuilder.of()
+                            .elementType(
+                                AttributeSetTypeBuilder.of()
+                                    .elementType(
+                                        AttributeNestedTypeBuilder.of()
+                                            .typeReference(
+                                                ProductTypeReferenceBuilder.of()
+                                                    .id(PRODUCT_TYPE_KEY_5)
+                                                    .build())
+                                            .build())
+                                    .build())
+                            .build())
+                    .build())
+            .isSearchable(false)
+            .isRequired(false)
+            .build();
+
+    final ProductTypeDraft oldProductTypeDraft =
+        ProductTypeDraftBuilder.of()
+            .key(PRODUCT_TYPE_KEY_5)
+            .name(PRODUCT_TYPE_NAME_5)
+            .description(PRODUCT_TYPE_DESCRIPTION_5)
+            .attributes(nestedTypeAttr)
+            .build();
+
+    // Sync productDraft with attribute referencing itself to source project
+    new ProductTypeSync(ProductTypeSyncOptionsBuilder.of(CTP_SOURCE_CLIENT).build())
+        .sync(List.of(oldProductTypeDraft))
+        .toCompletableFuture()
+        .join();
+    final ProductType oldProductType =
+        CTP_SOURCE_CLIENT
+            .productTypes()
+            .withKey(PRODUCT_TYPE_KEY_5)
+            .get()
+            .executeBlocking()
+            .getBody();
+
+    // test
+    final ProductTypeSync productTypeSync = new ProductTypeSync(productTypeSyncOptions);
+    final ProductTypeSyncStatistics productTypeSyncStatistics =
+        ProductTypeTransformUtils.toProductTypeDrafts(
+                CTP_SOURCE_CLIENT, referenceIdToKeyCache, List.of(oldProductType))
+            .thenCompose(newDrafts -> productTypeSync.sync(newDrafts))
+            .toCompletableFuture()
+            .join();
+
+    // assertion
+    assertThat(errorMessages).isEmpty();
+    assertThat(exceptions).isEmpty();
+    assertThat(builtUpdateActions).isEmpty();
+    assertThat(productTypeSyncStatistics).hasValues(1, 1, 0, 0, 0);
+    assertThat(productTypeSyncStatistics.getReportMessage())
```

Comment:
Can you add 2 more asserts: the product exists and it has the attribute that references itself?